### PR TITLE
feat(elements): init user provider

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

- add button styles
- add `<logto-user-provider>` to provide user context for elements

## Testing

image

image

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

**Review by wangsijie — APPROVED** [review]

File: packages/elements/index.html [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
   <meta charset="UTF-8">
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
   <title>Logto elements dev page</title>
   <script type="module" src="./src/index.ts"></script>
 </head>
+
 <body style="background: #111;">
   <logto-theme-provider theme="dark">
-    <logto-profile-card></logto-profile-card>
+    <logto-user-provider user='{"name": "Johnny Silverhand", "avatar": "https://github.com/logto-io.png"}'>
```
Comment (charIeszhao) [inline packages/elements/index.html:13]: Is "Johnny Silverhand" used for placeholder?

File: packages/elements/src/components/logto-button.styles.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  }
+
+  :host([size='large'][type='text']) {
+    height: 28px;
+    font: ${vars.fontLabel1};
+  }
+`;
+
+export const textButton = css`
+  :host([type='text']) {
```
Comment (charIeszhao) [inline packages/elements/src/components/logto-button.styles.ts:38]: Will the `:host` decorator impact the styles globally?

File: packages/elements/index.html (hunk 1 above)
Comment (gao-sun) [reply packages/elements/index.html:13]: it's removed in #6284

File: packages/elements/src/components/logto-button.styles.ts (hunk 2 above)
Comment (gao-sun) [reply packages/elements/src/components/logto-button.styles.ts:38]: you can double check MDN, it's kind of Web Components 101

File: packages/elements/index.html (hunk 1 above)
Comment (charIeszhao) [reply packages/elements/index.html:13]: OK, feels better that way

File: packages/elements/src/components/logto-button.styles.ts (hunk 2 above)
Comment (charIeszhao) [reply packages/elements/src/components/logto-button.styles.ts:38]: OK, will take a look

**Pull request merged**